### PR TITLE
html editor

### PR DESCRIPTION
**Pull request by dpolozov**

### CODE REVIEW

File: src/components/EditCampaign.js [hunk 1]
```diff
@@ -98,6 +103,7 @@ class EditCampaign extends React.Component {
     }
 
     async updateCampaign(){
+        console.log('update campaign called');
```
Comment (grishick) [inline src/components/EditCampaign.js]: nit: remove or comment out this log statement

File: src/components/EditCampaign.js [hunk 2]
```diff
@@ -310,7 +318,15 @@ class EditCampaign extends React.Component {
             maxAmount : maxAmount,
             currencyName: metaData.currencyName
         });
+        if(metaData.descriptionEditor){
+            setEditorState(metaData.descriptionEditor, true);
+            this.setState({updatedEditorState : true});
```
Comment (grishick) [inline src/components/EditCampaign.js:322]: what's the point of switching `updateEditorState` on/off?

File: src/components/TextEditor.js [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+        RichUtils.toggleInlineStyle(
+          this.state.editorState,
+          inlineStyle
+        )
+      );
+    }
+
+    render() {
+      if(!STOREDSTATE){
+        this.setState({STOREDSTATE});
```
Comment (grishick) [inline src/components/TextEditor.js:71]: doesn't this simply set empty stated? If so, why is this needed?

File: src/components/EditCampaign.js (hunk 1 above)
Comment (dpolozov) [reply src/components/EditCampaign.js]: removed

File: src/components/TextEditor.js [hunk 4]
```diff
@@ -0,0 +1,232 @@
...
+        )}
+      </div>
+    );
+  };
+
+  function getEditorState(){
+    return DESCRIPTIONRAW;
+  }
+
+  function setEditorState(storedState, hasContent){
```
Comment (grishick) [inline src/components/TextEditor.js:214]: hasContent seems unnecessary. Could the code simply check if there is anything inside storedState argument to decide if there is any content?

**Review by grishick — COMMENTED** [review]
Added some questions and small nits

File: src/components/EditCampaign.js (hunk 2 above)
Comment (dpolozov) [reply src/components/EditCampaign.js:322]: this was to make sure the data is loaded before the editor component is rendered. can't really update state or re-render the editor once it's there.

File: src/components/TextEditor.js (hunk 3 above)
Comment (dpolozov) [reply src/components/TextEditor.js:71]: the set has to be set to either the created empty from Editor or from existing. otherwise the component doesn't work

File: src/components/TextEditor.js (hunk 4 above)
Comment (dpolozov) [reply src/components/TextEditor.js:214]: not really. storedState is still a created editor object. i've tried a few things but it is a lot simpler to do that. there are editor content objects either way. would have to do a lot more complicated checking or just pass the boolean

**Review by grishick — APPROVED** [review]
